### PR TITLE
starting supporting queries for PirToRest

### PR DESCRIPTION
**Pull request by arcuri82**

### CODE REVIEW

**Review by arcuri82 — COMMENTED** [review]
@onurd86  please fix those issues i commented on

File: core-it/src/test/kotlin/org/evomaster/core/problem/rest/queries/QueriesTest.kt [hunk 1]
```diff
@@ -26,8 +30,246 @@ class QueriesTest : IntegrationTestRestBase(){
...
+        Assert.assertTrue((xParam.gene as OptionalGene).isActive)
+
+        // check that y is StringGene
+        val yParam = get.parameters.filter { it -> it.name.equals("y") }[0]
+        Assert.assertTrue(yParam.gene::class == StringGene::class)
+
+        // check that value of y is set to NOT_BAR
+        Assert.assertTrue(yParam.gene.getValueAsRawString().equals("NOT_BAR"))
+
+        // check that the status code is 200
```
[the remark this answers is not in the document]
Comment (onurd86) [reply core-it/src/test/kotlin/org/evomaster/core/problem/rest/queries/QueriesTest.kt]: Hi Andrea, 

Thanks for your comment. I have fixed that.

File: core-it/src/test/kotlin/org/evomaster/core/problem/rest/queries/QueriesTest.kt [hunk 2]
```diff
@@ -26,8 +30,246 @@ class QueriesTest : IntegrationTestRestBase(){
...
+        Assert.assertTrue(bParam.gene::class == DoubleGene::class)
+        Assert.assertTrue((bParam.gene as DoubleGene).value == 0.5)
+
+        // check that c is BooleanGene and its value is -0.2
+        val cParam = get.parameters.filter { it -> it.name.equals("c") }[0]
+        Assert.assertTrue(cParam.gene::class == BooleanGene::class)
+        Assert.assertTrue((cParam.gene as BooleanGene).value)
+
+        // the status code should be successful since we provided appropriate values
+        assertEquals(400, res.getStatusCode())
```
[the remark this answers is not in the document]
Comment (onurd86) [reply core-it/src/test/kotlin/org/evomaster/core/problem/rest/queries/QueriesTest.kt:203]: Hi Andrea, 

Thanks for your comment. I have fixed that.

File: core-it/src/test/kotlin/org/evomaster/core/problem/rest/queries/QueriesTest.kt [hunk 3]
```diff
@@ -26,8 +30,246 @@ class QueriesTest : IntegrationTestRestBase(){
...
+
+        // the value of b cannot be checked
+
+        // check that c is BooleanGene and its value is -0.2
+        val cParam = get.parameters.filter { it -> it.name.equals("c") }[0]
+        Assert.assertTrue(cParam.gene::class == BooleanGene::class)
+        Assert.assertTrue((cParam.gene as BooleanGene).value)
+
+        // the status code should be successful since we provided appropriate values
+        assertEquals(400, res.getStatusCode())
```
[the remark this answers is not in the document]
Comment (onurd86) [reply core-it/src/test/kotlin/org/evomaster/core/problem/rest/queries/QueriesTest.kt:231]: Hi Andrea, 

Thanks for your comment. I have fixed that.

**Pull request merged**